frontend/src/components/SignUpForm/SignUpForm.jsx

```diff
@@ -2,74 +2,105 @@ import { useState, useContext } from "react";
+
     const { updateUser } = useContext(UserContext);
     const navigate = useNavigate();
 
     const handleOnSubmit = async (event) => {
         event.preventDefault();
-        event.stopPropagation();
-        const userObj = {
-            firstName: firstName,
-            lastName: lastName,
-            email: email,
-            password: password
-        }
+        if (firstName && lastName && email && password) {
+          const isEmailValid = validateEmail(email);
+          const isPasswordValid = (password.length >= 8);
+          // Validate email and password
+          if (!isEmailValid && !isPasswordValid){
+            setEmailError("Invalid email address");
+            setPasswordError("Password must be at least 8 characters long");
+            return;
+          }
+            // Validate email
+            if (!isEmailValid) {
+              setEmailError("Invalid email address");
+            } else {
+              setEmailError(null);
+            }
+            // Validate password
+            if (!isPasswordValid) {
+              setPasswordError("Password must be at least 8 characters long");
+            } else {
+              setPasswordError(null);
+            }
+            if (!isEmailValid || !isPasswordValid){
+              return;
+            }
+          const userObj = {
+              firstName: firstName,
+              lastName: lastName,
+              email: email,
+              password: password
+          }
 
-        try {
-          // Make the signup API request
-          const response = await fetch(`http://localhost:3000/users`, {
-            method: 'POST',
-            headers: {
-              'Content-Type': 'application/json',
-            },
-            body: JSON.stringify(userObj),
-            credentials: 'include'
-          });
+          try {
+            // Make the signup API request
+            const response = await fetch(`http://localhost:3000/auth/registration`, {
+              method: 'POST',
+              headers: {
+                'Content-Type': 'application/json',
+              },
+              body: JSON.stringify(userObj),
+              credentials: 'include'
+            });
 
-          if (response.ok) {
-            const data = await response.json();
-            const loggedInUser = data.user;
+            if (response.ok) {
+              const data = await response.json();
+              const loggedInUser = data.user;
 
-            // Reset form fields
-            setFirstName('');
-            setLastName('');
-            setEmail('');
-            setPassword('');
+              // Reset form fields
+              setFirstName('');
+              setLastName('');
+              setEmail('');
+              setPassword('');
 
-            // Update the user context
-            updateUser(loggedInUser);
+              // Update the user context
+              updateUser(loggedInUser);
```

Comment:
pass data.user in directly, no need to put it into loggedInUser if you're only using it once